app/assets/javascripts/controllers/floating_ip/floating_ip_form_controller.js

```diff
@@ -1,75 +1,86 @@
-ManageIQ.angular.app.controller('floatingIpFormController', ['$http', '$scope', 'floatingIpFormId', 'miqService', function($http, $scope, floatingIpFormId, miqService) {
+ManageIQ.angular.app.controller('floatingIpFormController', ['$scope', 'floatingIpFormId', 'miqService', 'API', function($scope, floatingIpFormId, miqService, API) {
   var vm = this;
-  vm.floatingIpModel = { floating_ip_address: '' };
-  vm.formId = floatingIpFormId;
-  vm.afterGet = false;
-  vm.modelCopy = angular.copy( vm.floatingIpModel );
-  vm.model = "floatingIpModel";
 
-  ManageIQ.angular.scope = vm;
-  vm.saveable = miqService.saveable;
+  var init = function() {
+    vm.afterGet = false;
 
-  if (floatingIpFormId == 'new') {
-    vm.floatingIpModel.floating_ip_address = "";
-    vm.floatingIpModel.description = "";
-    vm.newRecord = true;
-  } else {
-    miqService.sparkleOn();
+    vm.floatingIpModel = { address: null };
+    vm.available_ems = null;
+
+    vm.formId = floatingIpFormId;
+    vm.model = "floatingIpModel";
+    vm.newRecord = floatingIpFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    vm.saveUrl = vm.newRecord ? '/floating_ip/create/new' : '/floating_ip/update/' + floatingIpFormId;
 
-    $http.get('/floating_ip/floating_ip_form_fields/' + floatingIpFormId)
-      .then(getFloatingIpFormData)
-      .catch(miqService.handleFailure);
-  }
+    vm.fields = [
+      'address',
+      'cloud_network_id',
+      'cloud_tenant_id',
+      'ems_id',
+      'network_port.ems_ref',
+    ];
+
+    miqService.sparkleOn();
+    if (vm.newRecord) {
+      API.get("/api/providers?expand=resources&attributes=name&filter[]=type='*NetworkManager'").then(function(data) {
```

Comment:
If https://github.com/ManageIQ/manageiq-ui-classic/pull/3736 is merged first, use `miqService.networkProviders()` here.